src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TelemetryBuffer.cs

```diff
@@ -103,6 +128,17 @@ public virtual void Process(ITelemetry item)
 
             lock (this)
             {
+                if (this.transmissionBuffer.Count >= this.MaximumUnsentBacklogSize)
+                {
+                    if (!this.itemDroppedMessageLogged)
+                    {
+                        TelemetryChannelEventSource.Log.ItemDroppedAsMaximumUnsentBacklogSizeReached(this.MaximumUnsentBacklogSize);
+                        this.itemDroppedMessageLogged = true;
+                    }
+
+                    return;
```

Comment:
We cannot use this kind of solution in the ServerTelemetryChannel since it is trying to guarantee that no items will be dropped.  Dropping items in this channel defeats the whole purpose of its existence.